include/internal/request_state.h

```diff
@@ -26,19 +26,22 @@ typedef struct {
   bool pending;
   bool match;
   uint16_t msg_id;
-  uint8_t compare_data[SBP_PAYLOAD_SIZE_MAX];
-  uint8_t compare_data_len;
+  settings_buf_t compare_data[SETTINGS_BUFLEN];
+  settings_buflen_t compare_data_len;
```

Comment:
TBH I'm not really sure the best way to handle this, I really think that `size_t` serves the purpose of a typedef for holding values that represent allocated buffer sizes. If anything `SETTINGS_BUFLEN` to me seems like the key centralizing factor here (and the implementation of it throughout this library). Holding cstrings in `uint8_t` arrays doesn't really bother me, since I'm pretty sure casting ASCII in between that and `char` shouldn't cause any issues.

Reply:
My pondering last night produced pretty similar thoughts, so I'll revert `settings_buf_t` and `settings_buflen_t` since they are pretty awkward.